### PR TITLE
[WIP] Query builder refactoring

### PR DESCRIPTION
**Pull request by Piezoid**
Rewrite QueryBuilder to use python `str.format()` engine and iterator.
I will rebase and redo the PR this on master if the refactoring branch is merged.

### CODE REVIEW

**Comment by Piezoid** [comment]
Sadly it needs Python 3.3 in order to use `yield from`...

**Comment by abretaud** [comment]
I see no objection to require python 3.3, can you update travis.yml and README.md accordingly?
I'll have a look at the whole PR later (today I hope)

File: askomics/libaskomics/rdfdb/ResultsBuilder.py [hunk 1]
```diff
@@ -55,11 +55,13 @@ def organize_attribute_and_entity(self, results, constraints): # FIXME results i
                 entity_attribute_list[entity_var][attr_var] = attribute_name
 
         #removing all empty node. means they are category...
-        for elem in entity_attribute_list.values():
-            for elt_att in elem:
-                if elt_att in  entity_attribute_list:
-                    del entity_attribute_list[elt_att]
-                    del entity_name_list[elt_att]
+        remove_item = [ elt_att
```
Comment (abretaud) [inline askomics/libaskomics/rdfdb/ResultsBuilder.py]: Just for curiosity, is there some benefit (perf?) in using this syntax?

Comment (Piezoid) [inline askomics/libaskomics/rdfdb/ResultsBuilder.py]: No, I came up with this while trying to understood the logic, but I couldn't find a better way to express this.  My previous code was changing the dict shape while iterating on it, causing a error.
I can revert to original if functional style is a problem.
Ideally categories should be treated separately...

File: askomics/libaskomics/rdfdb/SparqlQueryBuilder.py [hunk 2]
```diff
@@ -12,29 +15,42 @@ class SparqlQueryBuilder(ParamManager):
...
-    def load_from_file(self, template, replacement={}):
+    def load_from_file(self, template_file, replacement={}):
         """ Get a sparql query from a file, possibly replacing a template word by another given as argument """
-        #f = open('SPARQL_queries.txt', 'a')
-        #Add HEADER
-        query = self.header_sparql_config()
-        query += '\n'.join([line.rstrip('\r\n') for line in open(template)])
-        for joker in replacement.keys():
-            query = query.replace(joker, replacement[joker])
+        #FIXME: Callers should use the new naming convention (no '#')
```
Comment (abretaud) [inline askomics/libaskomics/rdfdb/SparqlQueryBuilder.py]: Why does it change (replaced #)?

File: askomics/libaskomics/rdfdb/SparqlQueryBuilder.py [hunk 3]
```diff
@@ -49,189 +65,116 @@ def get_neighbours_source_node(self, json, node_id):
...
-        return list_nodes
-
-    def more_than_one_child_with_the_same_uri(self, lneib):
-        for c in lneib:
-            for d in lneib:
-                if c['uri'] == d['uri'] and c['id'] != d['id']:
-                    return True
-        return False
-
-    def build_new_json_by_unionpath(self, json, node_id, lneib): # FIXME lneib is unused
```
Comment (abretaud) [inline askomics/libaskomics/rdfdb/SparqlQueryBuilder.py]: Can you keep it (and add a FIXME deadcode) as it is used by the dead get_block_request method?

**Comment by abretaud** [comment]
I think it's ok (apart from my comments), it will only need to be rebased once #4 is merged
If you get bored, unit tests would be great also
Thanks!

**Comment by abretaud** [comment]
This is replaced by #7

**Pull request closed without merging**